src/widget/wlabel.cpp

```diff
@@ -63,6 +65,34 @@ void WLabel::setup(QDomNode node, const SkinContext& context) {
             setAlignment(Qt::AlignLeft | Qt::AlignVCenter);
         }
     }
+    // Ellipsis
+    if (context.hasNode(node, "Ellipsis")) {
+        if (context.selectString(node, "Ellipsis").toLower() == "right") {
+            m_elideMode = Qt::ElideRight;
+        } else if (context.selectString(node, "Ellipsis").toLower() == "center") {
+            m_elideMode = Qt::ElideMiddle;
+        } else if (context.selectString(node, "Ellipsis").toLower() == "middle") {
+            m_elideMode = Qt::ElideMiddle;
+        } else if (context.selectString(node, "Ellipsis").toLower() == "left") {
+            m_elideMode = Qt::ElideLeft;
+        }
+    }
+}
+
+QString WLabel::text() const {
+    return m_longText;
+}
+
+void WLabel::setText(const QString& text) {
+    m_longText = text;
+    if (m_elideMode != Qt::ElideNone) {
+        QFontMetrics metrics(font());
+        // Measure the text for label width - 2 for 1 pixel border
```

Comment:
then this could be font-related... for now don't try to guess that it's an issue with a border, just say that we have to adjust the width so the entire thing appears.  But we also need to test on windows and mac to make sure we don't need to subtract even more